src/main/java/no/nav/folketrygdloven/beregningsgrunnlag/perioder/ErbruttoinntektForGradertAndelLikNull.java

```diff
@@ -0,0 +1,63 @@
+package no.nav.folketrygdloven.beregningsgrunnlag.perioder;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import no.nav.folketrygdloven.beregningsgrunnlag.regelmodell.AktivitetStatusV2;
+import no.nav.folketrygdloven.beregningsgrunnlag.regelmodell.AndelGradering;
+import no.nav.folketrygdloven.beregningsgrunnlag.regelmodell.BruttoBeregningsgrunnlag;
+import no.nav.folketrygdloven.beregningsgrunnlag.regelmodell.PeriodeModell;
+import no.nav.folketrygdloven.beregningsgrunnlag.regelmodell.PeriodisertBruttoBeregningsgrunnlag;
+
+class ErbruttoinntektForGradertAndelLikNull {
+
+    private ErbruttoinntektForGradertAndelLikNull(){
+    }
+
+    public static boolean vurder(PeriodeModell input, AndelGradering andelGradering, LocalDate dato){
+        for(PeriodisertBruttoBeregningsgrunnlag periodisertGrunnlag : input.getPeriodisertBruttoBeregningsgrunnlagList()){
+            if(periodisertGrunnlag.getPeriode().inneholder(dato)){
+                var grunnlag = finnMatchendeGrunnlag(periodisertGrunnlag.getBruttoBeregningsgrunnlag(), andelGradering);
+                if(grunnlag.isPresent()){
+                    return grunnlag.get().getBruttoBeregningsgrunnlag().equals(BigDecimal.ZERO);
+                } else{
+                    throw new IllegalStateException("Finner ikke matchende bruttoberegningsgrunnlag for andel som skal graderes");
+                }
+            }
+        }
+        return false;
+    }
+
+    private static Optional<BruttoBeregningsgrunnlag> finnMatchendeGrunnlag(List<BruttoBeregningsgrunnlag> beregningsgrunnlag, AndelGradering andelGradering){
+        if(andelErSnEllerFl(andelGradering)){
+            return beregningsgrunnlag.stream()
+                .filter(b -> b.getAktivitetStatus().equals(andelGradering.getAktivitetStatus()))
+                .findAny();
+        }
+        List<BruttoBeregningsgrunnlag> arbeidsforhold = beregningsgrunnlag.stream()
+            .filter(b -> b.getArbeidsforhold() != null)
+            .filter(b -> arbeidsforholdErPrivatpersonEllerOrganisasjon(b, andelGradering))
+            .collect(Collectors.toList());
+
+        if(arbeidsforhold.size() > 1){
+            return arbeidsforhold.stream()
+                .filter(b -> b.getArbeidsforhold().getArbeidsforholdId().equals(andelGradering.getArbeidsforhold().getArbeidsforholdId()))
+                .findAny();
+        }
+        return arbeidsforhold.isEmpty() ? Optional.empty() : Optional.of(arbeidsforhold.get(0));
```

Comment:
Eg forstår ikkje heilt kvifor dette blir riktig. Veit vi at arbeidsforholdet som graderer matcher arbeidsforholdet på det eksisterende beregningsgrunnlaget? Kva om det er gradering for eit nytt arbeidsforhold?